samcli/lib/samlib/resource_metadata_normalizer.py

```diff
@@ -67,6 +71,39 @@ def normalize(template_dict):
                     },
                 )
 
+        # This is a work around to allow the customer to use sam deploy or package commands without the need to provide
+        # values for the CDK auto generated asset parameters. The suggested solution is to let CDK add some metadata to
+        # the autogenerated parameters, so sam can skip them safely.
+
+        # Normalizing CDK auto generated parameters. Search for parameters that meet these conditions:
+        # 1- parameter name matches pattern
+        #    `AssetParameters[0-9a-f]{64}(?:S3Bucket|S3VersionKey|ArtifactHash)[0-9A-F]{8}`
+        # 2- parameter type is string
+        # 3- there is no reference to this parameter any where in the template resources
+        # 4- there is no default value for this parameter
+        # We set an empty string as default value for the matching parameters, so the customer can use sam deploy or
+        # package commands without providing values for the auto generated parameters, as these parameters are not used
+        # in SAM (sam set the resources paths directly, and does not depend on template parameters)
+        if normalize_parameters and is_cdk_project(template_dict):
+            resources_as_string = json.dumps(resources)
+            parameters = template_dict.get("Parameters", {})
+
+            default_value = " "
+            cdk_asset_parameter_pattern = re.compile(
+                "AssetParameters[0-9a-f]{64}(?:S3Bucket|S3VersionKey|ArtifactHash)[0-9A-F]{8}"
```

Comment:
👏🏼  One verification; first set of chars are lower case (since it is `[0-9a-f]`) and second set of chars are upper case (since it is `[0-9A-F]`)?
And do we want to keep this at file level so that regex won't be compiled every time we call this method?

Reply:
Although all generated Asset parameters I saw in CDK examples are following this pattern (the first 64 characters contain only small letters, and last 8 characters contain only capital letters), but I found that in CDK repo they are using [this pattern](https://github.com/aws/aws-cdk/blob/b1ecd3d49d7ebf97a54a80d06779ef0f0b113c16/packages/%40aws-cdk/assert-internal/lib/canonicalize-assets.ts#L19), so I updated the pattern and moved to file level.